### DISCUSSION TITLE
Can we safely change name/email attributes on the shop model?

### DISCUSSION THREAD

**Discussion by frankiejarrett** [Kyon147/laravel-shopify#25] [Q&A]
I'd  like to make the attributes on my `User` shop model reflect what is actually from the shop.

By default, `$user->name` is always `shop-domain.com` and `$user->email` is always `shop@shop-domain.com`.

Is it safe to update these attributes without affecting any package functionality?

**Basic concept**

```php
class User extends Authenticatable implements ShopModelInterface
{
    use ShopModel;

    protected static function booted()
    {
        static::created(function ($model) {
            dispatch(function () use ($model) {
                $response = $model->api()->rest('GET', '/admin/shop.json')['body']['shop'] ?? [];

                $model->updateQuietly([
                    'name' => $response['shop_owner'] ?? $model->name,
                    'email' => $response['email'] ?? $model->email,
                ]);
            });
        });
    }
}
```

**Comment by Kyon147** [comment]
The name on the user model is the `myshopify` admin url - this should not be changed as it is used in the app to get the domain.

The `email` field should be fine to change.
